src/tests/java/htsjdk/samtools/CRAMIndexQueryTest.java

```diff
@@ -0,0 +1,464 @@
+        final String referenceFileName,
+        final QueryInterval interval,
+        final String[] expectedNames) throws IOException
+    {
+        doQueryTest(
+            reader -> reader.queryContained(new QueryInterval[]{interval}),
+            cramFileName,
+            referenceFileName,
+            expectedNames
+        );
+    }
+
+    @Test(dataProvider="singleIntervalContained")
+    public void testQueryContainedSequence(
+        final String cramFileName,
+        final String referenceFileName,
+        final QueryInterval interval,
+        final String[] expectedNames) throws IOException
+    {
+        doQueryTest(
+            reader -> reader.queryContained(
+                reader.getFileHeader().getSequence(interval.referenceIndex).getSequenceName(),
+                interval.start,
+                interval.end
+            ),
+            cramFileName,
+            referenceFileName,
+            expectedNames
+        );
+    }
+
+    @Test(dataProvider="singleIntervalContained")
+    public void testQuerySingleIntervalContainedTrue(
+            final String cramFileName,
+            final String referenceFileName,
+            final QueryInterval interval,
+            final String[] expectedNames) throws IOException
+    {
+        doQueryTest(
+            reader -> reader.query(new QueryInterval[]{interval}, true),
+            cramFileName,
+            referenceFileName,
+            expectedNames
+        );
+    }
+
+    @Test(dataProvider="singleIntervalContained")
+    public void testQuerySequenceContainedTrue(
+            final String cramFileName,
+            final String referenceFileName,
+            final QueryInterval interval,
+            final String[] expectedNames) throws IOException
+    {
+        doQueryTest(
+            reader -> reader.query(
+                reader.getFileHeader().getSequence(interval.referenceIndex).getSequenceName(),
+                interval.start,
+                interval.end,
+                true
+            ),
+            cramFileName,
+            referenceFileName,
+            expectedNames
+        );
+    }
+
+    @DataProvider(name = "multipleIntervalOverlapping")
+    public Object[][] multipleIntervalOverlapping() {
+        return new Object[][]{
+            //{bamQueryWithBAI, null,
+            //        new QueryInterval[]{new QueryInterval(0, 100010, 100010), new QueryInterval(0, 100011, 100011)},
+            //        new String[]{"a", "b", "c", "d", "e"}},
+            {cramQueryWithCRAI, cramQueryReference,
+                    new QueryInterval[]{new QueryInterval(0, 100010, 100010), new QueryInterval(0, 100011, 100011)},
+                    new String[]{"a", "b", "c", "d", "e"}},
+            {cramQueryWithBAI, cramQueryReference,
+                    new QueryInterval[]{new QueryInterval(0, 100010, 100010), new QueryInterval(0, 100011, 100011)},
+                    new String[]{"a", "b", "c", "d", "e"}},
```

Comment:
Can also try porting the test cases from `ReadsDataSourceUnitTest` here, which use the `reads_data_source_test1.bam` file mentioned above.